katdal/dataset.py

```diff
@@ -1018,14 +1019,29 @@ def lst(self):
         """
         return self.sensor['Antennas/%s/lst' % self.ref_ant] * (12 / np.pi)
 
+    def _sensor_per_ant(self, base_name):
+        """Extract a single sensor per antenna and safely stack the results."""
+        def sensor_data(ant_name):
+            return self.sensor['Antennas/%s/%s' % (ant_name, base_name)]
+        return np.column_stack([sensor_data(ant.name) for ant in self.ants]) \
+            if self.ants else np.zeros((self.shape[0], 0))
+
+    def _sensor_per_corrprod(self, base_name):
+        """Extract a single sensor per corrprod and safely stack the results."""
+        def sensor_data(antA, antB):
+            return self.sensor['Antennas/%s/%s_%s' % (antA, base_name, antB)]
+        return np.column_stack([sensor_data(inpA[:-1], inpB[:-1])
+                                for inpA, inpB in self.corr_products]) \
+            if self.corr_products else np.zeros((self.shape[0], 0))
```

Comment:
Thank you for picking this up. Of course I only checked on the problematic corner case and not the normal files... I really need some unit tests here!